include/infrastructure.php

```diff
@@ -87,21 +59,24 @@ function create_custom_relevance() {
 
 //---------------------------------------------------------------------------//
 function create_reversal_tables () {
-	global $wpdb;
 
 	$table = REVERSALTABLE;
-	$sql = "CREATE TABLE `" . $table . "` (
-		`id` varchar(50) NOT NULL,
-		`language_code` varchar(30) CHARACTER SET " . COLLATION . " COLLATE " . FULLCOLLATION . ",
-		`reversal_head` longtext CHARACTER SET " . COLLATION . " COLLATE " . FULLCOLLATION . ",
-		`reversal_content` longtext CHARACTER SET " . COLLATION . " COLLATE " . FULLCOLLATION . ",
-		`sortorder` INT NOT NULL DEFAULT '0',
-		`browseletter` varchar(5) CHARACTER SET " . COLLATION . " COLLATE " . FULLCOLLATION .
-		$sql .= ", UNIQUE KEY (`id`)";
-		$sql .= ") CHARACTER SET " . COLLATION . " COLLATE " . FULLCOLLATION . ";";
-
-	//echo $sql . "<br>";
-	require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
+	$char_set = MYSQL_CHARSET;
+	$collate = MYSQL_COLLATION;
+
+	$sql = <<<SQL
+CREATE TABLE `{$table}` (
+  `id` VARCHAR(50) NOT NULL,
+  `language_code` VARCHAR(30),
+  `reversal_head` LONGTEXT,
+  `reversal_content` LONGTEXT,
+  `sortorder` INT NOT NULL DEFAULT '0',
+  `browseletter` VARCHAR(5),
+  UNIQUE KEY idx_{$table}_id (`id`)
+) DEFAULT CHARSET={$char_set} COLLATE={$collate};
+SQL;
+
+	include_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
```

Comment:
Most of the time I prefer to use `include` rather than `require` because if there is a problem in the required file, PHP just stops execution without returning any debug information. With `include` the exception can be caught and handled.